comp110-worksheet-6/OxoBoard.cs

```diff
@@ -1,76 +1,160 @@
-
-					switch (GetSquare(x, y))
-					{
-						case Mark.None:
-							Console.Write(" "); break;
-						case Mark.O:
-							Console.Write("O"); break;
-						case Mark.X:
-							Console.Write("X"); break;
-					}
-				}
-
-				Console.WriteLine();
-			}
-		}
-	}
+    public enum Mark { None, O, X };
+
+    public class OxoBoard
+    {
+        private int BoardWidth;
+        private int BoardHeight;
+        private Mark[,] Board;
+        private int InARow;
+        // TODO:
+        private readonly List<int[,]> WinConditions = new List<int[,]> {
+             new int [3,2] { { 0, 0 }, { 0, 1 }, { 0, 2 } },
+             new int [3,2] { { 1, 0 }, { 1, 1 }, { 1, 2 } },
+             new int [3,2] { { 2, 0 }, { 2, 1 }, { 2, 2 } },
+             new int [3,2] { { 0, 0 }, { 1, 0 }, { 2, 0 } },
+             new int [3,2] { { 0, 1 }, { 1, 1 }, { 2, 1 } },
+             new int [3,2] { { 0, 2 }, { 1, 2 }, { 2, 2 } },
+             new int [3,2] { { 0, 0 }, { 1, 1 }, { 2, 2 } },
+             new int [3,2] { { 2, 0 }, { 1, 1 }, { 0, 2 } },
+        };
+
+        // Constructor. Perform any necessary data initialisation here.
+        // Uncomment the optional parameters if attempting the stretch goal -- keep the default values to avoid breaking unit tests.
+        public OxoBoard(int width = 3, int height = 3, int inARow = 3)
+        {
+            BoardWidth = width;
+            BoardHeight = height;
+            Board = new Mark[BoardWidth, BoardHeight];
+            InARow = inARow;
+        }
+
+        // Return the contents of the specified square.
+        public Mark GetSquare(int x, int y)
+        {
+            return Board[x, y];
+        }
+
+        // If the specified square is currently empty, fill it with mark and return true.
+        // If the square is not empty, leave it as-is and return False.
+        public bool SetSquare(int x, int y, Mark mark)
+        {
+            bool result = Board[x, y] == Mark.None ? true : false;
+            if (result) Board[x, y] = mark;
+            return result;
+        }
+
+        // If there are still empty squares on the board, return false.
+        // If there are no empty squares, return true.
+        public bool IsBoardFull()
+        {
+            for (int x = 0; x < /*BoardWidth*/3; x++)
+            {
+                for (int y = 0; y < /*BoardHeight*/3; y++)
+                {
+                    if (GetSquare(x, y) == Mark.None)
+                        return false;
+                }
+            }
+            return true;
+        }
+
+        private List<Mark> GetMarks(int[,] positions)
+        {
+            List<Mark> result = new List<Mark>(3);
+            for(int i = 0; i < 3; i++)
+            {
+                int x = positions[i,0];
+                int y = positions[i,1];
+                result.Add(GetSquare(x, y));
+            }
+            return result;
+        }
+
+        private bool Every<T>(List<T> list, Func<T, T, bool> predicate)
```

Comment:
I think you could have got the same functionality using builtin methods from the `System.Linq` namespace